#! /bin/sh /usr/share/dpatch/dpatch-run
## 50_parse-inline-brackets.dpatch by Rafael Laboissiere <rafael@debian.org>
##
## DP: Fix a parsing problem in function handles and inlines were brackets
## DP: are present.  This patch has been applied upstream in CVS. See
## DP: https://www.cae.wisc.edu/pipermail/bug-octave/2006-October/000903.html 

@DPATCH@

--- octave2.9-2.9.9.orig/src/pt-cell.cc
+++ octave2.9-2.9.9/src/pt-cell.cc
@@ -110,7 +110,7 @@
     {
       tree_argument_list *elt = *p;
 
-      append (elt ? elt->dup (sym_tab) : 0);
+      new_cell->append (elt ? elt->dup (sym_tab) : 0);
     }
 
   new_cell->copy_base (*this);
--- octave2.9-2.9.9.orig/src/pt-mat.cc
+++ octave2.9-2.9.9/src/pt-mat.cc
@@ -942,7 +942,7 @@
     {
       tree_argument_list *elt = *p;
 
-      append (elt ? elt->dup (sym_tab) : 0);
+      new_matrix->append (elt ? elt->dup (sym_tab) : 0);
     }
 
   new_matrix->copy_base (*this);
